### DISCUSSION TITLE
mobility

### DISCUSSION THREAD

**Discussion by roeysalah** [omnetpp/omnetpp#1089] [Q&A]
hello everyone
I'm new to omnet++ ,and I'm trying to build a simulation where a host sends messages to itself and every time it receives a self-message it will move in the simulation according to the values ​​it received in the message
I would appreciate it if you could help me build such a simulation

**Comment by rhornig** [comment]
This is called mobility model in INET. See https://inet.omnetpp.org/docs/users-guide/ch-mobility for further info. 

OMNeT++ itself does not define what is the 'position' of the node, so you may decide how to implement such a thing. The most straightforward approach is to define two member variables (x,y) in each node and set those variables what you receive the self message. After that you have to implement also the `refreshDisplay()` method and set the display string position tags in it based on your x,y variables.

Also take a look at the (somewhat more complex) Canvas sample.

**Reply by roeysalah** [reply]
thank you , i will try to implement  the refreshDisplay()
